parquet-hadoop/src/test/java/org/apache/parquet/format/converter/TestParquetMetadataConverter.java

```diff
@@ -580,6 +583,42 @@ private void testBinaryStats(StatsHelper helper) {
     Assert.assertTrue(roundTripStats.isEmpty());
   }
 
+  @Test
+  public void testBinaryStatsWithTruncation() {
+    int defaultTruncLen = ParquetProperties.DEFAULT_STATISTICS_TRUNCATE_LENGTH;
+    int[] validLengths = {1, 2, 16, 64, defaultTruncLen - 1, defaultTruncLen - 1};
+    for (int len : validLengths) {
+      testBinaryStatsWithTruncation(len);
+    }
+
+    int[] invalidLengths = {-1, 0, defaultTruncLen + 1};
+    for (int len : invalidLengths) {
+      try {
+        testBinaryStatsWithTruncation(len);
+      } catch (IllegalArgumentException e) {
+        // expected, nothing to do
+      }
+    }
+  }
+
+  private void testBinaryStatsWithTruncation(int truncateLen) {
```

Comment:
Please add test data where we would reach the size limit of statistics.

Reply:
Added it.

Reply:
added